tests/test_mock_sampler.py

```diff
@@ -216,7 +216,79 @@ def test_yield_arguments(self):
                                    broken_edges=delete_edges)
         self.assertTrue(len(sampler.nodelist)==4)
         self.assertTrue(len(sampler.edgelist)==1)
-        
+
+    def test_custom_mock_sampler(self):
+        """Test that MockDWaveSampler uses the provided custom mocking_sampler."""
+
+        # Define a custom sampler that always returns the same sample
+        class CustomSampler(dimod.Sampler):
+            properties = {}
+            parameters = {}
+
+            def sample(self, bqm, **kwargs):
+                # Return a sample where all variables are set to 1
+                sample = {v: 1 for v in bqm.variables}
+                energy = bqm.energy(sample)
+                return dimod.SampleSet.from_samples_bqm(sample, bqm)
+
+        custom_sampler = CustomSampler()
+
+        # Create a simple BQM
+        bqm = dimod.BQM({'a': -1, 'b': -1}, {('a', 'b'): -1}, 0.0, vartype='SPIN')
```

Comment:
At this problem size, the exact solver would be triggered. should override the default `exact_solver_cutoff`

Reply:
Another option is to return two samples (num_reads=2) and check the second sample. This might be simpler.